utils/add-attribution.sh

```diff
@@ -16,10 +16,41 @@
 set -o errexit -o nounset -o pipefail
 
 PKG_PATH="${1:-.}"
-BLUEPRINT_NAME="${BLUEPRINT_NAME:-$(grep -m 1 "name:" "${PKG_PATH}/Kptfile" | sed 's/[[:space:]]*name:[[:space:]]*//')}"
-BLUEPRINT_VERSION="${BLUEPRINT_VERSION:-$(cd "${PKG_PATH}" && git describe --abbrev=0)}"
 
-kpt fn source "${PKG_PATH}" |
-    kpt cfg grep "apiVersion=cnrm" |
-    kpt fn run --image gcr.io/kpt-fn/set-annotation:v0.1 -- "cnrm.cloud.google.com/blueprint=cnrm/${BLUEPRINT_NAME}/${BLUEPRINT_VERSION}" |
-    kpt fn sink "${PKG_PATH}"
+CATALOG_PATH="catalog"
+BLUEPRINT_TOOL="cnrm"
+BLUEPRINT_ANNOTATION="cnrm.cloud.google.com/blueprint"
+
+KPT_VERSION="$(kpt version)"
+if [[ "${KPT_VERSION}" != "1."* || "${KPT_VERSION}" == "1.0.0-beta.1" || "${KPT_VERSION}" == "1.0.0-alpha"* ]]; then
+    echo >&2 "Error: requires kpt verion of at least 1.0.0-beta.2"
+    exit 2
+fi
+
+function print_catalog_path() {
+    local path="${1}"
+    if [[ ! -f "${path}/Kptfile" ]]; then
+        echo >&2 "Error: no Kptfile in package path: ${path}"
+        return 2
+    fi
+    local full_path="$(cd "${path}" && pwd -P)"
+    local repo_root="$(cd "${full_path}" && git rev-parse --show-toplevel)"
+    local prefix="${repo_root}/${CATALOG_PATH}/"
+    if [[ "${full_path}" != "${prefix}"* ]]; then
+        echo >&2 "Error: package not in catalog: ${path}"
+        return 2
+    fi
+    local relative_path="${full_path#"${prefix}"}"
+    echo "${relative_path}"
+}
+
+function print_repo_version() {
+    local path="${1}"
+    echo "$(cd "${path}" && git describe --abbrev=0)"
+}
+
+BLUEPRINT_NAME="${BLUEPRINT_NAME:-$(print_catalog_path "${PKG_PATH}")}"
+BLUEPRINT_VERSION="${BLUEPRINT_VERSION:-$(print_repo_version "${PKG_PATH}")}"
+BLUEPRINT_ATTRIBTUION="${BLUEPRINT_TOOL}/${BLUEPRINT_NAME}/${BLUEPRINT_VERSION}"
+
+kpt fn eval "${PKG_PATH}" --image gcr.io/kpt-fn/set-annotation:v0.1 -- "${BLUEPRINT_ANNOTATION}=${BLUEPRINT_ATTRIBTUION}"
```

Comment:
Based on [this comment}(https://github.com/GoogleContainerTools/kpt/issues/2423#issuecomment-890253553) we should use `set-annotations` instead.